src/current/v25.2/detect-hotspots.md

```diff
@@ -0,0 +1,141 @@
+  - If **No**, and the metrics outlier appears in a metric other than CPU percent or latch conflict wait duration, consider causes other than a hotspot.
+
+## Step 2. Check for existence of `no split key found` log
+
+The [`no split key found` log]({% link {{ page.version.version }}/load-based-splitting.md %}#monitor-load-based-splitting) is emitted in the [`KV_DISTRIBUTION` log channel]({% link {{ page.version.version }}/logging-overview.md %}#logging-channels). This log is only emitted when a single replica begins using a significant percentage of the resources on the node where it resides.
+
+This log is not associated with a specific event type but includes an unstructured message, for example:
+
+```
+I250523 21:59:25.755283 31560 13@kv/kvserver/split/decider.go:298 ⋮ [T1,Vsystem,n5,s5,r1115/3:‹/Table/106/1/{113338-899841…}›] 2979  no split key found: insufficient counters = 0, imbalance = 20, most popular key occurs in 36% of samples, access balance right-biased 98%, popular key detected, clear direction detected
+```
+
+In the preceding log example, the square-bracketed tag section provides the following information:
+
+- node ID: `n5` indicates the node ID is 5.
+- range ID: `r1115` indicates the range ID is 1115.
+
+The timestamp at the beginning of the log is `250523 21:59:25.755283`.
+
+The unstructured message ends with one of the following string combinations:
+
+1. `popular key detected, clear direction detected`
+1. `popular key detected, no clear direction`
+1. `no popular key, clear direction detected`
+1. `no popular key, no clear direction`
+
+### A. `popular key detected`
+
+The `popular key detected` log indicates that a significant percentage of reads or writes target a single row within a range.
+
+- To check for a `popular key detected` log, search the `KV_DISTRIBUTION` logs on the hot node from Step 1 within the noted time period.
+
+- Once you identify a relevant log, note the range ID in the tag section of the log.
+
+- If the outlier appears in the latch conflict wait durations metric, does a `popular key detected` log exist?
+
+  - If **Yes**, it may be a [write hotspot]({% link {{ page.version.version }}/understand-hotspots.md %}#write-hotspot). Note the range ID of `popular key detected` log and proceed to find the corresponding [hot ranges log](#step-3-find-hot-ranges-log).
+  - If **No**, investigate other reasons for the latch conflict wait durations metric outlier.
+
+- If the outlier appears in the CPU percent metric, does a `popular key detected` log exist?
+
+  - If **Yes**, it may be a [read hotspot]({% link {{ page.version.version }}/understand-hotspots.md %}#read-hotspot), because the write hotspot was ruled out with the latch conflict wait durations metric. The order of operations in this troubleshooting process matters. Note the range ID of `popular key detected` log and proceed to find the corresponding [hot ranges log](#step-3-find-hot-ranges-log).
```

Comment:
possible typo, this currently says "latch conflict wait durations metric", did you mean to edit this to match "CPU percent metric" in the previous bullet?